grpc/server/src/engine_server/mappings/ipc/upgrade_request.rs

```diff
@@ -51,6 +51,37 @@ impl TryFrom<UpgradeRequest> for UpgradeConfig {
             )
         };
 
+        let new_auction_delay: Option<u64> = if !upgrade_point.has_new_auction_delay() {
```

Comment:
How does it work if we upgrade the auction delay? E.g. if we change from 3 to 5, will the next auction determine three eras instead of just one?

Reply:
I won't lie, I am not exactly sure

Reply:
I'm just wary of making things configurable or even upgradeable if changing their config value or upgrading them doesn't work.

Reply:
The EE upgrade process has been in place for about a year now. How the auction contract will behave if these values change, on the other hand, should be defined (if it isn't already) and tests should be added to enforce it. 

However, this story / PR is just about moving the constants to the chainspec.

@darthsiroftardis please follow up with Henry, Michal, and Alexander to make sure they are aware the situation, but remain focused on your scoped work.

Reply:
Will do!

Reply:
Changing either of these two delays will also require special handling in the consensus component, though.
I agree that this shouldn't be part of this PR, but let's make sure we have tickets for designing the required communication with and handling in the consensus component.

Reply:
@afck: you might be the best person to write those tickets, as they are consensus-related :wink:

Reply:
I created [NDRS-551](https://casperlabs.atlassian.net/browse/NDRS-551) for the consensus side; we probably also need one for the auction contract?

I'd still be in favor of removing those two fields from `UpgradeConfig` instead. It might be a long time until we can actually schedule implementing NDRS-551, and until then, using those fields in an upgrade would probably just break the system.

Reply:
I suspect that we are basically addressing this concern over here: https://github.com/CasperLabs/casper-node/pull/403#discussion_r513737996

Reply:
OK, then let's keep these fields in `UpgradeConfig`, I guess, and make sure we implement it before we use it.